packages/machine-extractor/src/extractAction.ts

```diff
@@ -0,0 +1,411 @@
+      eventObject = {
+        type: 'expression',
+        value: fileContent.slice(arg2.start!, arg2.end!),
+      };
+    }
+
+    // Options
+    if (t.isObjectExpression(arg3)) {
+      // delay = string | number | (ctx, e) => string | number
+      const foundDelay = arg3.properties.find(
+        prop =>
+          t.isObjectProperty(prop) &&
+          t.isIdentifier(prop.key) &&
+          prop.key.name === 'delay',
+      ) as t.ObjectProperty | undefined;
+      if (foundDelay) {
+        if (
+          t.isArrowFunctionExpression(foundDelay.value) ||
+          t.isFunctionExpression(foundDelay.value)
+        ) {
+          delay = {
+            type: 'expression',
+            value: fileContent.slice(foundDelay.start!, foundDelay.end!),
+          };
+        } else if (
+          t.isStringLiteral(foundDelay.value) ||
+          t.isNumericLiteral(foundDelay.value)
+        ) {
+          delay = {
+            type: getLiteralType(foundDelay.value),
+            value: foundDelay.value.value,
+          };
+        }
+      }
+
+      // id = string
+      const foundId = arg3.properties.find(
+        prop =>
+          t.isObjectProperty(prop) &&
+          t.isIdentifier(prop.key) &&
+          prop.key.name === 'id',
+      ) as t.ObjectProperty | undefined;
+      if (
+        foundId &&
+        (t.isStringLiteral(foundId.value) || t.isNumericLiteral(foundId.value))
+      ) {
+        id = {
+          type: getLiteralType(foundId.value),
+          value: foundId.value.value,
+        };
+      }
+    }
+
+    return {
+      event: eventObject,
+      actorRef: actorRef!,
+      options: {
+        delay,
+        id,
+      },
+    };
+  }
+
+  throw Error(`Unsupported sendTo expression`);
+}
+
+function getLiteralType(value: t.ObjectProperty['value']) {
+  if (
+    t.isNullLiteral(value) ||
+    t.isRegExpLiteral(value) ||
+    t.isTemplateLiteral(value)
+  ) {
+    throw Error('null,regexp or template literals can not have literal type');
+  }
+  if (t.isNumericLiteral(value)) {
+    return 'number';
+  }
+  if (t.isStringLiteral(value)) {
+    return 'string';
+  }
+  if (t.isBooleanLiteral(value)) {
+    return 'boolean';
+  }
+  throw Error('Unsupported literal property value');
+}
+
+export const isBuiltinActionWithName = (actionNode: ActionNode, name: string) =>
+  t.isCallExpression(actionNode.node) &&
+  t.isIdentifier(actionNode.node.callee) &&
+  actionNode.node.callee.name === name;
+
+function extractEventObject(
+  eventObject: t.ObjectExpression,
+  fileContent: string,
+): Record<string, { type: JsonEntryType; value: JsonValue }> {
+  const extracted: Record<
+    string,
+    { type: JsonEntryType; value: JsonValue }
+  > = {};
+  eventObject.properties.forEach(prop => {
+    if (t.isObjectProperty(prop)) {
+      if (t.isIdentifier(prop.key)) {
+        if (t.isLiteral(prop.value)) {
+          if (
+            t.isRegExpLiteral(prop.value) ||
+            t.isTemplateLiteral(prop.value) ||
```

Comment:
expression-less template literals, like this one:
```ts
`just a string`
```
should be treated as a string, not as an expression